### PR TITLE
Create and accept MPoS block (QTUMCORE 67 and 68)

### PR DESCRIPTION
**Pull request by timemarkovqtum**
The reward will be divided into 10 recipients, 9 of the recipients are block creators that previously created block.
The block miner that created a block submitted script for receiving reward after 500 blocks (generated coins maturity - COINBASE_MATURITY) with length of 10 blocks.
If a recipient created more blocks in the period of 10 blocks, then will receive more of the outputs when the block reward is divided. 

```
Block creation parameter:
BlockReward = 4;
TxFee = 2;
Refund = 1;
Input0 = 160;

PoS outputs:
Output0 = NULL;
Output1 = 165;
Output2 = 1;

PoS outputs:
Output0 = NULL;
Output1 = 160.5;
Output2 = 0.5;
Output3 = 0.5;
Output4 = 0.5;
Output5 = 0.5;
Output6 = 0.5;
Output7 = 0.5;
Output8 = 0.5;
Output9 = 0.5;
Output10 = 0.5;
Output11 = 1;
```

Algorithm:
1) Block is created and the reward is split into 10 recipients
2) 9 scripts from previous block creators that have the required age read and list is created
3) List of scripts is maintained in order to avoid reading the blocks again
4) The block is checked that the reword is split as required and the scripts to the recipients are the one that have the required age

### CODE REVIEW

**Comment by Earlz** [comment]
@timemarkovqtum Can you rebase so that it can be merged onto master soon?

**Comment by timemarkovqtum** [comment]
@Earlz It is re-based.

**Comment by timemarkovqtum** [comment]
The staker send the public key in order for the users be able to verify the digital signature of the block, the digital signature cannot be verified without the public key, so need to be present in 1 output.
The other 9 outputs for the reward in the MPoS block can be public key hash, so that modification can be performed.

**Comment by Earlz** [comment]
In this it's only possible to split a single stake into 2 outputs. This should not be a consensus rule, stakers should be capable of splitting a stake into as many outputs as they want, as long as the MPoS participants outputs are in the block and they still receive their appropriate reward amount

**Comment by timemarkovqtum** [comment]
I am working on parameterization for stake split into more then 2 outputs.
Can you update the example in QTUMCORE-68 in order to have the exact result that is required?
Please confirm that is also required adding 2 new parameters outside the consensus parameters that the user can pass to the staker for `split threshold` and `split outputs`, for example: `split threshold = 100`, `split outputs = 5`, the stake UTXO will be split into 5 outputs if bigger then 500.

File: src/net_processing.cpp [hunk 1]
```diff
@@ -3347,6 +3347,14 @@ bool ProcessNetBlock(const CChainParams& chainparams, const std::shared_ptr<cons
     {
         LOCK(cs_main);
 
+        // Check that the coinstake transaction exist in the received block
+        if(pblock->IsProofOfStake() && !(pblock->vtx.size() > 1 && pblock->vtx[1]->IsCoinStake()))
```
Comment (Earlz) [inline src/net_processing.cpp]: I think this is good to be paranoid about security, but was there a specific reason this was added?

**Comment by Earlz** [comment]
@qtum-neil and @timemarkovqtum please review my changes. We will need to fix --disable-wallet mode later in a separate PR

**Comment by Earlz** [comment]
This was tested and confirmed to properly handle a mix of PoW and PoS blocks in regtest, as well as verified to give proper payouts

File: src/net_processing.cpp (hunk 1 above)
Comment (timemarkovqtum) [inline src/net_processing.cpp]: This is extra check on the entry point where all blocks are received.

**Comment by timemarkovqtum** [comment]
@Earlz Testing: I make quick testing on the code, the results are the same as before.
Code check: The refund outputs are ignored when do the check, so can be between the rewards, if no other check. nRefundVouts is not used but present.

**Comment by Earlz** [comment]
@qtum-neil please give final sign off on this. I've tested it and it works as expected for me

**Pull request merged**